plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,834 @@
+    mcp = module.params.get("mcp")
+    pfc_admin_state = module.params.get("pfc_admin_state")
+    access_macsec_policy = module.params.get("access_macsec_policy")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template)
+    mso_template.validate_template("fabricPolicy")
+
+    path = "/fabricPolicyTemplate/template/interfacePolicyGroups"
+    object_description = "Interface Policy Groups"
+
+    template_info = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {})
+
+    existing_interface_policies = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("interfacePolicyGroups", [])
+    if interface_policy_group or interface_policy_group_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_interface_policies,
+            [KVPair("uuid", interface_policy_group_uuid) if interface_policy_group_uuid else KVPair("name", interface_policy_group)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_interface_policies
+
+    if state == "present":
+
+        if match:
+            if interface_policy_group and match.details.get("name") != interface_policy_group:
+                ops.append(dict(op="replace", path="{0}/{1}/name".format(path, match.index), value=interface_policy_group))
+                match.details["name"] = interface_policy_group
+
+            if description is not None and match.details.get("description") != description:
+                ops.append(dict(op="replace", path="{0}/{1}/description".format(path, match.index), value=description))
+                match.details["description"] = description
+
+            if interface_type and match.details.get("type") != interface_type:
+                mso.fail_json(msg="Interface type cannot be changed.")
+
+            if domains:
+                # the existing domain list is updated by the new domain list
+                domain_uuid = []
+                # get names and uuids of existing domains
+                existing_physical_domains = {domain["name"]: domain["uuid"] for domain in template_info.get("domains", [])}
+                existing_l3_domains = {domain["name"]: domain["uuid"] for domain in template_info.get("l3Domains", [])}
+                for item in domains:
+                    if item in existing_physical_domains:
+                        domain_uuid.append(existing_physical_domains[item])
+                    elif item in existing_l3_domains:
+                        domain_uuid.append(existing_l3_domains[item])
+                    else:
+                        mso.fail_json(msg="Domain '{0}' not found in the template '{1}'.".format(item, template))
+                if set(domain_uuid) != set(match.details.get("domains")):
```

Comment:
```suggestion
                if set(domain_uuid) != set(match.details.get("domains") if match.details.get("domains") else []):
```
`match.details.get("domains")` can return `None` and you would end up with `TypeError: 'NoneType' object is not iterable`

Reply:
changed